src/transformers/data/data_collator.py

```diff
@@ -142,3 +142,21 @@ def mask_tokens(self, inputs: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]
 
         # The rest of the time (10% of the time) we keep the masked input tokens unchanged
         return inputs, labels
+
+
+@dataclass
+class DataCollatorForQuestionAnswering(DataCollator):
+    """
+    Data collator used for language modeling.
+    - collates batches of tensors
+    - preprocesses batches for question answering
+    """
+
+    def collate_batch(self, batch: List[Dict[str, torch.Tensor]]) -> Dict[str, torch.Tensor]:
+        keys = batch[0].keys()
+        inputs = {}
+
+        for key in keys:
+            inputs[key] = torch.stack([example[key] for example in batch])
+
+        return inputs
```

Comment:
pretty clean implementation of data collator. cc @sgugger

Reply:
Yes, quick question though, why does the default not work here?

Reply:
(I think it's ~identical to the default one so maybe we can merge them and just use the default here)

Reply:
default collator needs `List[InputDataClass]` but I'm returning List[Dict`[str, torch.Tensor]]` so I was not able to use the default one

Reply:
InputDataClass is just a type alias for anything.

Reply:
>InputDataClass is just a type alias for anything.

Yes, but I think it assumes that it will be class as it uses `getattr` and `vars`. How should I proceed ?

Reply:
maybe we can change the default implem to accept more input types

Reply:
Let me know if you want to do it @sgugger or @patil-suraj

Reply:
@julien-c can we just add a simple type check in default collator, i.e if the input is `dict` we can call `example.get` instead of `getattr`, or maybe convert the `dict` to a simple `class` ?

Reply:
I'll add this this morning with the backward compatibility to the old `DataCollator` style.